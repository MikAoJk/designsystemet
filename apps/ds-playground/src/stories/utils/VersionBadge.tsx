import { ReactNode } from 'react';

interface VersionBadgeProps {
  packageName: string;
}

export const VersionBadge = ({ packageName }: VersionBadgeProps): ReactNode => {
  let color = 'green';

  let version = process.env[`@skatteetaten/ds-${packageName}`] ?? '..';
  if (process.env['NODE_ENV'] === 'development') {
    color = 'red';
    version = 'developing';
  }
  if (process.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    color = 'blue';
  }

  const normalizedPackageName = packageName.replaceAll('-', '--');
  const normalizedVersion = version.replaceAll('-', '--').replaceAll('_', '__');

  return (
    <img
      src={`https://img.shields.io/badge/%40skatteetaten%2Fds--${normalizedPackageName}-${normalizedVersion}-${color}?logo=npm`}
      alt={`@skatteetaten/ds-${packageName}@${version}`}
    />
  );
};
