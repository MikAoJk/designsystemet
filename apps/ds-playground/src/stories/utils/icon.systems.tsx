import { SystemSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const SystemSVGPaths = findSVGPathElements(SystemSVGPathsAndIcons);
export const SystemIconLabels = makeLabelsFromSVGPathElements(SystemSVGPaths);
