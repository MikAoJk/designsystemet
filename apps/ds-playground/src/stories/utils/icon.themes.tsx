import { ThemeSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const ThemeSVGPaths = findSVGPathElements(ThemeSVGPathsAndIcons);
export const ThemeIconLabels = makeLabelsFromSVGPathElements(ThemeSVGPaths);
