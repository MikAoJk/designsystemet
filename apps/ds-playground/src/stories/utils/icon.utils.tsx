import {
  SystemSVGPathsAndIcons,
  ThemeSVGPathsAndIcons,
} from '@skatteetaten/ds-icons';

export type SVGPathAndIconType =
  | typeof SystemSVGPathsAndIcons
  | typeof ThemeSVGPathsAndIcons;
export type SVGPathAndIconKey = keyof SVGPathAndIconType;

export const makeLabelsFromSVGPathElements = function (
  SVGPathElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, string> {
  const labels: Record<string, string> = {};
  Object.keys(SVGPathElements).forEach((name: string) => {
    labels[name] = name.replace('SVGpath', '');
  });

  return labels;
};

export const findSVGPathElements = function (
  SVGPathAndIconElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, SVGPathAndIconType[SVGPathAndIconKey]> {
  const elements: Record<string, SVGPathAndIconType[SVGPathAndIconKey]> = {};
  Object.keys(SVGPathAndIconElements).forEach((elementName: string) => {
    if (elementName.endsWith('SVGpath')) {
      elements[elementName] =
        SVGPathAndIconElements[elementName as SVGPathAndIconKey];
    }
  });

  return elements;
};
