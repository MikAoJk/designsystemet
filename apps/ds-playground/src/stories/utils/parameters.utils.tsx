import { ArgsTable, Stories, PRIMARY_STORY } from '@storybook/addon-docs';
import type { Parameters } from '@storybook/csf';

type DesignParameters = {
  name: string;
  url: string;
};

export const getParameters = function (
  designParameters: DesignParameters[]
): Parameters {
  return {
    docs: {
      page: (): JSX.Element => {
        return (
          <>
            <Stories title={''} />
            <ArgsTable story={PRIMARY_STORY} />
          </>
        );
      },
    },
    design: designParameters.map((designParameter: DesignParameters) => {
      return {
        name: designParameter.name,
        type: 'figma',
        url: designParameter.url,
      };
    }),
    backgrounds: {
      values: [
        { name: 'Svart', value: '#000' },
        { name: 'Hvit', value: '#fff' },
      ],
    },
  };
};
