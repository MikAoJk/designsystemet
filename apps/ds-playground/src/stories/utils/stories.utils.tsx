export const exampleParameters = {
  docs: {
    source: {
      type: 'code',
      language: 'tsx',
      transform: (source: string): string => {
        return source
          .replace('render: (_args): JSX.Element', 'const Examples = ()')
          .replace(
            'render: function Render(_args): JSX.Element ',
            'const Examples = () => '
          )
          .replace('satisfies Story', '');
      },
    },
  },
  /* _args i en story må finnes slik at props i fanen Controls skjules og er prefikset med underscore for å unngå eslint klaging */
  controls: {
    exclude: /.*/,
  },
};
