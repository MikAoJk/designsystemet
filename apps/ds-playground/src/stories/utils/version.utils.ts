export const getVersion = (
  packageName: string
): {
  major: string;
  minor: string;
  patch: string;
  postfix?: string;
  style?: Record<string, string>;
} => {
  const ver = process.env[`@skatteetaten/${packageName}`] ?? '..';
  const [major, minor, patch] = ver.split('.');
  const majorWithPackageName = packageName ? `${packageName}@${major}` : major;

  const style = {
    color: 'var(--semantic-page-foreground)',
    background: 'var(--semantic-success-background)',
    'font-weight': 'var(--font-weight-bold)',
    'font-size': '13px',
  };

  if (process.env['NODE_ENV'] === 'development') {
    return {
      major: majorWithPackageName,
      minor,
      patch,
      postfix: 'develop',
      style: {
        ...style,
        background: 'var(--semantic-danger-background)',
      },
    };
  }
  if (process.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    return {
      major: majorWithPackageName,
      minor,
      patch,
      postfix: process.env['STORYBOOK_BUILD_VERSION'],
      style: {
        ...style,
        background: 'var(--semantic-warning-background)',
      },
    };
  }

  return {
    major: majorWithPackageName,
    minor,
    patch,

    style: {
      ...style,
    },
  };
};
