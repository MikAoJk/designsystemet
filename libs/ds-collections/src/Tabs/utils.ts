export const valueRegex = /^[\w:.-_]*$/;
