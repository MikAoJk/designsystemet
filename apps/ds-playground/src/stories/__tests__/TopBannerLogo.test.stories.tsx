import { dsI18n } from '@skatteetaten/ds-core-utils';
import { expect } from '@storybook/jest';
import { Meta, StoryObj } from '@storybook/react';
import { within } from '@storybook/testing-library';

// eslint-disable-next-line @nx/enforce-module-boundaries
import { wrapper } from './testUtils/storybook.testing.utils';
// eslint-disable-next-line @nx/enforce-module-boundaries
import { TopBannerLogo } from '../../../../../libs/ds-layout/src/TopBannerLogo/TopBannerLogo';
// eslint-disable-next-line @nx/enforce-module-boundaries
import { TopBannerLogoProps } from '../../../../../libs/ds-layout/src/TopBannerLogo/TopBannerLogo.types';
import customLogo from '../../assets/custom-logo.svg';
import customMobileLogo from '../../assets/custom-mobile-logo.svg';

const meta = {
  component: TopBannerLogo,
  title: 'Tester/TopBannerLogo (intern)',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    logo: {
      table: { disable: true },
      control: 'select',
      options: ['', customLogo],
    },
    mobileLogo: {
      table: { disable: true },
      control: 'select',
      options: ['', customMobileLogo],
    },
    noLinkLogo: { table: { disable: true } },
    // HTML
    alt: { table: { disable: true } },
    href: { table: { disable: true } },
  },
} satisfies Meta<typeof TopBannerLogo>;
export default meta;
type Story = StoryObj<typeof meta>;

const frontPageText = dsI18n.t('ds_pages:topbannerexternal.SkeLogoLinkText');
const logoText = dsI18n.t('ds_pages:topbannerexternal.SkeLogoImageText');
const defaultArgs: TopBannerLogoProps = {};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ref: (instance: HTMLDivElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'en',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toHaveAttribute('id', 'htmlId');
    await expect(container).toHaveClass('dummyClassname');
    await expect(container).toHaveAttribute('lang', 'en');
    await expect(container).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Defaults (A8, A9)',
  args: {
    ...defaultArgs,
  },
  argTypes: {},
  parameters: {
    imageSnapshot: {
      focus: `${wrapper} a`,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const link = canvas.getByRole('link');
    await expect(link).toBeInTheDocument();
    await expect(link).toHaveAttribute('href', 'https://www.skatteetaten.no/');
    const logo = canvas.getByRole('img', {
      name: frontPageText,
    });
    await expect(logo).toBeInTheDocument();
  },
} satisfies Story;

export const WithNoLinkLogo = {
  name: 'With NoLinkLogo (A8)',
  args: {
    ...defaultArgs,
    noLinkLogo: true,
  },
  argTypes: {
    noLinkLogo: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    // eslint-disable-next-line testing-library/no-node-access
    const link = canvasElement.querySelector('a');
    await expect(link).not.toBeInTheDocument();
    const logo = canvas.getByRole('img', {
      name: logoText,
    });
    await expect(logo).toBeInTheDocument();
  },
} satisfies Story;

export const WithLogoMobileLogoAltAndHref = {
  name: 'With Logo MobileLogo Alt And Href (A8, A10)',
  args: {
    ...defaultArgs,
    logo: customLogo,
    mobileLogo: customMobileLogo,
    alt: 'custom alt text',
    href: '#',
  },
  argTypes: {
    logo: { table: { disable: false } },
    mobileLogo: { table: { disable: false } },
    alt: { table: { disable: false } },
    href: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const link = canvas.getByRole('link', {
      name: 'custom alt text',
    });
    await expect(link).toBeInTheDocument();
    await expect(link).toHaveAttribute('href', '#');
  },
} satisfies Story;

export const WithBreakpointXS = {
  name: 'With Breakpoint-xs',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-xs',
    },
  },
} satisfies Story;

export const WithBreakpointS = {
  name: 'With Breakpoint-s',
  args: {
    ...defaultArgs,
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-s',
    },
  },
} satisfies Story;
