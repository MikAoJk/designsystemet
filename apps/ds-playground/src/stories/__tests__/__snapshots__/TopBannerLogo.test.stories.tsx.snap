// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/TopBannerLogo (intern) Defaults play-test 1`] = `
<div data-test-block="true">
  <div class="TopBannerLogo_containerLogo__HNnbt">
    <a href="https://www.skatteetaten.no/"
       class="TopBannerLogo_logoLink__DvORV"
    >
      <img class="TopBannerLogo_logo_mobile__x4lpB"
           alt="Forside Skatteetaten"
           src="ske-logo-mobile.c19c0351f0931cd93ae02f37a7fb7498.svg"
      >
      <img class="TopBannerLogo_logo_desktop__SI+eK"
           alt="Forside Skatteetaten"
           src="ske-logo.efcae54a4ea6b0ab43b2b65cd3ca166f.svg"
      >
    </a>
  </div>
</div>
`;

exports[`Tester/TopBannerLogo (intern) WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div id="htmlId"
       class="dummyClassname TopBannerLogo_containerLogo__HNnbt"
       lang="en"
       data-testid="123ID"
  >
    <a href="https://www.skatteetaten.no/"
       class="TopBannerLogo_logoLink__DvORV"
    >
      <img class="TopBannerLogo_logo_mobile__x4lpB"
           alt="Forside Skatteetaten"
           src="ske-logo-mobile.c19c0351f0931cd93ae02f37a7fb7498.svg"
      >
      <img class="TopBannerLogo_logo_desktop__SI+eK"
           alt="Forside Skatteetaten"
           src="ske-logo.efcae54a4ea6b0ab43b2b65cd3ca166f.svg"
      >
    </a>
  </div>
</div>
`;

exports[`Tester/TopBannerLogo (intern) WithBreakpointM smoke-test 1`] = `
<div data-test-block="true">
  <div class="TopBannerLogo_containerLogo__HNnbt">
    <a href="https://www.skatteetaten.no/"
       class="TopBannerLogo_logoLink__DvORV"
    >
      <img class="TopBannerLogo_logo_mobile__x4lpB"
           alt="Forside Skatteetaten"
           src="ske-logo-mobile.c19c0351f0931cd93ae02f37a7fb7498.svg"
      >
      <img class="TopBannerLogo_logo_desktop__SI+eK"
           alt="Forside Skatteetaten"
           src="ske-logo.efcae54a4ea6b0ab43b2b65cd3ca166f.svg"
      >
    </a>
  </div>
</div>
`;

exports[`Tester/TopBannerLogo (intern) WithLogoMobileLogoAltAndHref play-test 1`] = `
<div data-test-block="true">
  <div class="TopBannerLogo_containerLogo__HNnbt">
    <a href="#"
       class="TopBannerLogo_logoLink__DvORV"
    >
      <img class="TopBannerLogo_logo_mobile__x4lpB"
           alt="custom alt text"
           src="custom-mobile-logo.e7c1e07901657fb09f8543357f7fa867.svg"
      >
      <img class="TopBannerLogo_logo_desktop__SI+eK"
           alt="custom alt text"
           src="custom-logo.252414682ce8a77d1dfc266bb47a0e0e.svg"
      >
    </a>
  </div>
</div>
`;

exports[`Tester/TopBannerLogo (intern) WithMobileScreen smoke-test 1`] = `
<div data-test-block="true">
  <div class="TopBannerLogo_containerLogo__HNnbt">
    <a href="https://www.skatteetaten.no/"
       class="TopBannerLogo_logoLink__DvORV"
    >
      <img class="TopBannerLogo_logo_mobile__x4lpB"
           alt="Forside Skatteetaten"
           src="ske-logo-mobile.c19c0351f0931cd93ae02f37a7fb7498.svg"
      >
      <img class="TopBannerLogo_logo_desktop__SI+eK"
           alt="Forside Skatteetaten"
           src="ske-logo.efcae54a4ea6b0ab43b2b65cd3ca166f.svg"
      >
    </a>
  </div>
</div>
`;

exports[`Tester/TopBannerLogo (intern) WithNoLinkLogo play-test 1`] = `
<div data-test-block="true">
  <div class="TopBannerLogo_containerLogo__HNnbt">
    <img class="TopBannerLogo_logo_mobile__x4lpB"
         alt="Skatteetaten logo"
         src="ske-logo-mobile.c19c0351f0931cd93ae02f37a7fb7498.svg"
    >
    <img class="TopBannerLogo_logo_desktop__SI+eK"
         alt="Skatteetaten logo"
         src="ske-logo.efcae54a4ea6b0ab43b2b65cd3ca166f.svg"
    >
  </div>
</div>
`;

exports[`Tester/TopBannerLogo (intern) WithRef play-test 1`] = `
<div data-test-block="true">
  <div class="TopBannerLogo_containerLogo__HNnbt"
       id="dummyIdForwardedFromRef"
  >
    <a href="https://www.skatteetaten.no/"
       class="TopBannerLogo_logoLink__DvORV"
    >
      <img class="TopBannerLogo_logo_mobile__x4lpB"
           alt="Forside Skatteetaten"
           src="ske-logo-mobile.c19c0351f0931cd93ae02f37a7fb7498.svg"
      >
      <img class="TopBannerLogo_logo_desktop__SI+eK"
           alt="Forside Skatteetaten"
           src="ske-logo.efcae54a4ea6b0ab43b2b65cd3ca166f.svg"
      >
    </a>
  </div>
</div>
`;
